tests/utils/test_parsers.py

```diff
@@ -0,0 +1,364 @@
+import unittest
+
+from epcpy import (
+    epc_pure_identity_to_gs1_element,
+    epc_pure_identity_to_gs1_element_string,
+    epc_pure_identity_to_gs1_key,
+    epc_pure_identity_to_gs1_keyed,
+    epc_pure_identity_to_scheme,
+    epc_pure_identity_to_tag_encodable,
+    hex_to_tag_encodable,
+    ignore_errors,
+    tag_uri_to_tag_encodable,
+)
+from epcpy.epc_schemes.adi import ADI
+from epcpy.epc_schemes.bic import BIC
+from epcpy.epc_schemes.cpi import CPI
+from epcpy.epc_schemes.gdti import GDTI
+from epcpy.epc_schemes.giai import GIAI
+from epcpy.epc_schemes.gid import GID
+from epcpy.epc_schemes.ginc import GINC
+from epcpy.epc_schemes.grai import GRAI
+from epcpy.epc_schemes.gsin import GSIN
+from epcpy.epc_schemes.gsrn import GSRN
+from epcpy.epc_schemes.gsrnp import GSRNP
+from epcpy.epc_schemes.imovn import IMOVN
+from epcpy.epc_schemes.itip import ITIP
+from epcpy.epc_schemes.pgln import PGLN
+from epcpy.epc_schemes.sgcn import SGCN
+from epcpy.epc_schemes.sgln import SGLN
+from epcpy.epc_schemes.sgtin import GTIN_TYPE, SGTIN
+from epcpy.epc_schemes.sscc import SSCC
+from epcpy.epc_schemes.upui import UPUI
+from epcpy.epc_schemes.usdod import USDOD
+from epcpy.utils.common import ConvertException
+
+VALID_TEST_DATA = [
+    {
+        "scheme": ADI,
+        "tag_encodable": True,
+        "gs1_keyed": False,
+        "gs1_element": False,
```

Comment:
The information in these 3 fields is already included in the `scheme` class right?  Using `issubclass` you could get the same behavior in the tests below.

Reply:
Definitely, but I wanted to make this test independent of the actual scheme implementation since a missing subclass definition for a scheme would skip the test entirely

Reply:
Ok fair point!